classification_analysis.ipynb

```diff
@@ -605,26 +708,56 @@
     "        trips = test_trip if type(test_trip) is list else [test_trip]\n",
     "    TP, FN, FP, TN = {}, {}, {}, {}\n",
     "    for trip in trips:\n",
+    "        gt_trajectory = trip['gt_trajectory']\n",
     "        ss_timeline, gt_timeline = align_timelines(trip)\n",
     "        for mode in set(BASE_MODE.values()):\n",
     "            for ss in ss_timeline:\n",
     "                for gts in gt_timeline:\n",
     "                    if ss['end_ts'] >= gts['start_ts'] and ss['start_ts'] <= gts['end_ts']:\n",
-    "                        dur = min(ss['end_ts'], gts['end_ts']) - max(ss['start_ts'], gts['start_ts'])\n",
+    "                        range_start = max(ss['start_ts'], gts['start_ts'])\n",
+    "                        range_end = min(ss['end_ts'], gts['end_ts'])\n",
+    "\n",
+    "                        filtered_gt_distance = gt_trajectory.loc[\n",
+    "                            (gt_trajectory.ts >= range_start) & (gt_trajectory.ts <= range_end), :\n",
+    "                        ]\n",
+    "\n",
+    "                        if filtered_gt_distance.shape[0] == 0:\n",
+    "                            dist = 0\n",
```

Comment:
Does this happen a lot? I would *not* expect that to happen and we might want to assert instead

Reply:
It happens on some occasions, yes.

Reply:
@rahulkulhalli  can we print out the trip/section for which this happens (in addition to setting `dist=0`)? I don't think this should happen, and I would like to verify against the stored reference trajectories for the specific use cases.